src/ui/public/modals/modal_overlay.js

```diff
@@ -5,16 +5,23 @@ import modalOverlayTemplate from './modal_overlay.html';
  * Appends the modal to the dom on instantiation, and removes it when destroy is called.
  */
 export class ModalOverlay {
-  constructor(modalElement) {
+  constructor(modalElement, scope) {
```

Comment:
> I think we should only bind ESC key functionality if showClose is true in confirm_modal

I disagree. Keyboard navigation is key, and ESC means "no" on a confirmation dialog. If we're going to replace native functionality, we need to provide the same interactions in our replacement.

> if we don't show the X then I think we should force the user to be explicit about their choice

We are. They can't do anything else until they deal with the confirm dialog. "no" is a valid response, and ESC allows the user to respond that way with from the keyboard, just like they could always do before.

The functionality we're trying to replicate here is that of `window.confirm`, and adding ESC handling gets us there.